pkg/controller/job/job_controller.go

```diff
@@ -341,16 +350,23 @@ func (jm *JobController) updateJob(old, cur interface{}) {
 	}
 }
 
-// obj could be an *batch.Job, or a DeletionFinalStateUnknown marker item.
-func (jm *JobController) enqueueController(job interface{}) {
-	key, err := controller.KeyFunc(job)
+// obj could be an *batch.Job, or a DeletionFinalStateUnknown marker item,
+// immediate tells the controller to update the status right away, and should
+// happen ONLY when there was a successful pod run.
+func (jm *JobController) enqueueController(obj interface{}, immediate bool) {
+	key, err := controller.KeyFunc(obj)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", job, err))
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", obj, err))
 		return
 	}
 
-	// Retrieves the backoff duration for this Job
-	backoff := getBackoff(jm.queue, key)
+	var backoff time.Duration
+	if immediate {
+		backoff = time.Duration(0)
```

Comment:
I think that forcing the backoff value to 0 will only do half of the job because it didn't clear the counter of the previous failures in the rateLimitedQueue.

what do you think about the following code?
```
if immediate {
  jm.queue.Forget(key)
}
backoff := getBackoff(jm.queue, key)
```

Reply:
sgtm